public/javascript/pump/socket.js

```diff
@@ -134,9 +149,34 @@
             }
         };
 
-        sock.onclose = function() {
-            // XXX: reconnect?
+        sock.onclose = function(e) {
             Pump.socket = null;
+            // Reconnect on broken connections, not for normal close
+            if (e.code !== 1000) {
```

Comment:
Where is this error code coming from?

Reply:
From sock.js https://github.com/sockjs/sockjs-client/blob/master/lib/main.js#L357, sock.js is a little undocumented

Reply:
:thumbsup: